packages/blockchain-wallet-v4-frontend/src/data/analytics/sagas.ts

```diff
@@ -47,21 +45,15 @@ export default ({
   }
 
   const generateUniqueUserID = function * () {
-    const defaultHDWallet = yield select(
-      selectors.core.wallet.getDefaultHDWallet
-    )
     yield call(waitForUserData)
     const userId = (yield select(
       selectors.core.kvStore.userCredentials.getUserId
     )).getOrFail()
     if (userId) return userId
-    const { seedHex } = defaultHDWallet
-    const mnemonic = BIP39.entropyToMnemonic(seedHex)
-    const masterhex = BIP39.mnemonicToSeed(mnemonic)
-    const masterHDNode = Bitcoin.HDNode.fromSeedBuffer(masterhex)
-    let hash = crypto.sha256('info.blockchain.matomo')
-    let purpose = hash.slice(0, 4).readUInt32BE(0) & 0x7fffffff
-    return masterHDNode.deriveHardened(purpose).getAddress()
+
+    const guid = yield select(selectors.core.wallet.getGuid)
+    let hash = crypto.sha256(guid).toString('base64')
+    return hash
```

Comment:
So this hash will be rehashed one more time before being store in matomo ? in L200 ?

Reply:
Discussed via slack